syntax = "proto3";

package Serialize;

message Header {
	int32 version = 1;
	int32 command = 2;
	int32 time = 3;
	int32 length = 4;
};

message RegRequest {
	string login = 1;
	string password = 2;
};

message InRequest {
	string login = 1;
	string password = 2;
};

message Request {
	oneof msg {
		RegRequest register_request = 1;
		InRequest input_request = 2;
	}
}

message WrappedMessage {
	oneof msg {
		Request request = 1;
	}
}
